File: src/1822.sign-of-the-product-of-an-array.1950/solution.ts
function arraySign (nums: number[]): number {
    let result = 1;
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] < 0) {
            result = -result;
        } else if (nums[i] === 0) {
            return 0;
        }
    }
    return result;
}

File: src/1822.sign-of-the-product-of-an-array.1950/solution.java
class Solution {
    public int arraySign(int[] nums) {
        int flag = 1;
        for (int i : nums) {
            if (i == 0) {
                return 0;
            }
            if (i < 0) {
                flag = -flag;
            }
        }
        return flag;
    }
}